app.js

```diff
@@ -1 +1,52 @@
-console.log("js file is connected")
\ No newline at end of file
+checkedRadio = document.querySelector("input[name='ingredient']:checked");
+searchIngredient(checkedRadio);
+
+/**
+ * This function adds and ingredient a url, that is used to 
+ * make the call to the API for birthday cakes with that ingredient
+ * 
+ * @param {object} radioInput radio input DOM Element
+ */
+function searchIngredient(radioInput) {
+  const rootUrl = "https://api.edamam.com/search";
+  const recipeUrlElement = document.querySelector("#recipe-url");
+  const labelElement = document.querySelector("#label");
+  const imageElement = document.querySelector("#cake-img");
+  const ingredientsListElement = document.querySelector(".ingredients");
```

Comment:
It's always a good practice to define the varialbles as close to the point you use them as possible. I would write it l like this:
```javascript
function searchIngredient(radioInput) {
    //you clear the ingredients list first 
    ingredientsListElement.innerHTML = "";
    //then prepare your url
    const rootUrl = "https://api.edamam.com/search";
    const apiUrl = `${rootUrl}?q=birthday+cake+${radioInput.value}&app_id=589ecbd6&app_key=6d6116bfcbdc60fe641222727dc9eb8f`;
    //and then you start your API-call with this url:
    fetch(apiUrl)
    ...
}
```
And you need the other variables (`recipeUrlElement`, `labelElement` etc) only later on. So there is no need to define them directly in the beginnig of your function. It reduces the readability of the code.

Reply:
I think you are right that a variable should be defined near the code, but I still think this are global variables, because the function is re-making them everytime it is call, as global they would be made once, and be used different times and they would be organized for future purposes at the top of the field.

Reply:
also because in a difference to other variables they are dom elements, if we were use them to anywhere else, we know where to find them, and we avoid remaking them all over.